### PR TITLE
[Core] Convert custom webpack cell-loader into Babel Plugin

### PR DESCRIPTION
**Pull request by RobertBroersma**
This implements #503 and opens up the possibility for using Cells in Jest!

I've never written a Babel Plugin or done anything with AST's before, so I just wanted to get this out here to gather some feedback before looking into writing tests for it, and also to discuss on where to locate the plugin file!

This also fixes some issues with Cells where they break if you comment out some const exports and supports using function declarations instead of variable function declarations!

### CODE REVIEW

File: packages/core/config/babel-plugin-redwood-cell.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                      false,
+                      true
+                    )
+                  )
+                ),
+              ])
+            )
+          )
+
+          exportNames = []
```
Comment (RobertBroersma) [inline packages/core/config/babel-plugin-redwood-cell.js:49]: This seems hacky, but I couldn't find another way to do this. Any suggestions?

File: packages/core/config/babel-plugin-redwood-cell.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+module.exports = function ({ types: t }) {
+  const names = [
```
Comment (peterp) [inline packages/core/config/babel-plugin-redwood-cell.js:2]:
```suggestion
  // These are the expected named exports from a Cell file.
  const EXPORTS_FROM_CELL = [
```

File: packages/core/config/babel-plugin-redwood-cell.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        let name
+        if (declaration.type === 'VariableDeclaration') {
+          name = declaration.declarations[0].id.name
+        }
+
+        if (declaration.type === 'FunctionDeclaration') {
+          name = declaration.id.name
+        }
+
+        if (names.includes(name)) {
```
Comment (peterp) [inline packages/core/config/babel-plugin-redwood-cell.js:64]:
```suggestion
        if (EXPORTS_FROM_CELL.includes(name)) {
```

File: packages/core/config/babel-plugin-redwood-cell.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+module.exports = function ({ types: t }) {
```
Comment (peterp) [inline packages/core/config/babel-plugin-redwood-cell.js:1]:
```suggestion
// This is supposed to wrap a file that's has a suffix of `Cell` in Redwood's `withCell` higher order component.
// The HOC is responsible for the lifecycle methods during a graphQL query.
// The end result of this plugin is something like:
// ```js
// import { withCell } from '@redwoodjs/web'
// export default withCell({ QUERY, Loading, Succes, Failure, Empty, beforeQuery, afterQuery })
// ```
module.exports = function ({ types: t }) {
```

File: packages/core/config/babel-plugin-redwood-cell.js (hunk 1 above)
Comment (peterp) [reply packages/core/config/babel-plugin-redwood-cell.js:49]: I think it's fine, maybe clarify why you're doing that with a comment.

One thing that might make this code read a bit easier is if you switched around `ExportNamedDeclaration` and `Program`

File: packages/core/config/babel-plugin-redwood-cell.js (hunk 2 above)
Comment (peterp) [reply packages/core/config/babel-plugin-redwood-cell.js:2]: Maybe `CELL_EXPORTS` is a better name 🤷‍♂️

Comment (RobertBroersma) [reply packages/core/config/babel-plugin-redwood-cell.js]: I like `EXPORTS_FROM_CELL` better!

File: packages/core/config/babel-plugin-redwood-cell.js (hunk 1 above)
Comment (RobertBroersma) [reply packages/core/config/babel-plugin-redwood-cell.js:57]: Good idea! Will do.

**Comment by RobertBroersma** [comment]
@peterp Thanks for the feedback! It's processed. 👍 

I've decided to name the variable what it is; `EXPECTED_EXPORTS_FROM_CELL`. Let me know if that doesn't make sense.

I also added some tests for the plugin. I noticed tests weren't running on the `core` package. Is there a reason for that? I configured them to run so I could run my plugin test.

File: packages/core/package.json [hunk 5]
```diff
@@ -33,6 +33,7 @@
     "babel-plugin-auto-import": "^1.0.5",
     "babel-plugin-graphql-tag": "^2.5.0",
     "babel-plugin-module-resolver": "^4.0.0",
+    "babel-plugin-tester": "^9.0.1",
```
Comment (peterp) [inline packages/core/package.json:36]: I think this should be added to the devDeps instead of deps (otherwise it's installed in every Redwood app.)

Comment (RobertBroersma) [reply packages/core/package.json:36]: Ah, you are right! Missed the `-D` there.

**Comment by thedavidprice** [comment]
@RobertBroersma great work all around here, Robert, from super helpful inline comments to ramping up tests for the core package! 🚀

**Pull request merged**